api/src/main/java/org/openmrs/module/metadatamapping/api/db/MetadataMappingDAO.java

```diff
@@ -130,4 +133,115 @@
 	 * @return list of matching metadata items
 	 */
 	<T extends OpenmrsMetadata> List<T> getMetadataItems(Class<T> type, String metadataSourceName);
+	
+	/**
+	 * Save a new metadata set or update an existing one.
+	 * @param metadataSet object to save
+	 * @return saved object
+	 */
+	MetadataSet saveMetadataSet(MetadataSet metadataSet);
+	
+	/**
+	 * Get metadata set with the given id.
+	 * @param metadataSetId database id of the object
+	 * @return object or null, if does not exist
+	 */
+	MetadataSet getMetadataSet(Integer metadataSetId);
+	
+	/**
+	 * Get metadata set with the given uuid. 
+	 * @param metadataSetUuid uuid of the object
+	 * @return object or null, if does not exist
+	 */
+	MetadataSet getMetadataSetByUuid(String metadataSetUuid);
+	
+	/**
+	 * Get metadata set with the given name. 
+	 * @param metadataSource source of the set
+	 * @param metadataSetCode code of the set   
+	 */
+	MetadataSet getMetadataSet(MetadataSource metadataSource, String metadataSetCode);
+	
+	/**
+	 * Save a new metadata set member or update an existing one.
+	 * @param metadataSetMember object to save
+	 * @return saved object
+	 * @see #saveMetadataSetMembers(Collection)
+	 */
+	MetadataSetMember saveMetadataSetMember(MetadataSetMember metadataSetMember);
+	
+	/**
+	 * Save a collection of new metadata set members or update an existing ones.
+	 * @param metadataSetMembers collection of objects to save
+	 * @return the same collection with saved objects
+	 * @see #saveMetadataSetMember(MetadataSetMember)
+	 */
+	Collection<MetadataSetMember> saveMetadataSetMembers(Collection<MetadataSetMember> metadataSetMembers);
+	
+	/**
+	 * Get metadata set member with the given id.
+	 * @param metadataSetMemberId database id of the object
+	 * @return object or null, if does not exist
+	 */
+	MetadataSetMember getMetadataSetMember(Integer metadataSetMemberId);
+	
+	/**
+	 * Get members of a metadata set. If members have {@link MetadataSetMember#getSortWeight()} set they will be ordered 
+	 * in ascending order according to said weight. Note that due to differences in database implementations, the order 
+	 * will be unpredictable, if there are null sort weights in the set.
```

Comment:
That is a good clarification. There are two different issues at hand here:
* null values might be sorted first or last depending on database 
vendor: This issue I have documented in the javadoc.
* row ordering may be inconsistent if order by clause is omitted: I may 
have omitted the order by clause in some cases where explicit ordering 
is not required. I will fix these. The javadoc in question is not 
related to this issue.